shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPluginTest.mm

```diff
@@ -0,0 +1,210 @@
+}
+
+- (NSString*)getStringRepresentationForRange:(NSRange)range {
+  return [NSString stringWithFormat:@"(%lu, %lu)", range.location, range.length];
+}
+
+#pragma mark UITextChecker Overrides
+
+- (NSRange)rangeOfMisspelledWordInString:(NSString*)stringToCheck
+                                   range:(NSRange)range
+                              startingAt:(NSInteger)startingOffset
+                                    wrap:(BOOL)wrapFlag
+                                language:(NSString*)language {
+  return self.startingIndexToRange[@(startingOffset)].rangeValue;
+}
+
+- (NSArray<NSString*>*)guessesForWordRange:(NSRange)range
+                                  inString:(NSString*)string
+                                  language:(NSString*)language {
+  return self.rangeToSuggestions[[self getStringRepresentationForRange:range]];
+}
+
+@end
+
+@interface FlutterSpellCheckPlugin ()
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result;
+- (UITextChecker*)textChecker;
+@end
+
+@interface FlutterSpellCheckPluginTest : XCTestCase
+
+@property(strong, nonatomic) id mockMethodChannel;
+@property(strong, nonatomic) FlutterSpellCheckPlugin* plugin;
+@property(strong, nonatomic) id mockTextChecker;
+
+@end
+
+#pragma mark -
+
+@implementation FlutterSpellCheckPluginTest
+
+- (void)setUp {
+  [super setUp];
+  self.mockMethodChannel = OCMClassMock([FlutterMethodChannel class]);
+  self.plugin = [[FlutterSpellCheckPlugin alloc] initWithMethodChannel:self.mockMethodChannel];
+  __weak FlutterSpellCheckPlugin* weakPlugin = self.plugin;
+  OCMStub([self.mockMethodChannel invokeMethod:[OCMArg any]
+                                     arguments:[OCMArg any]
+                                        result:[OCMArg any]])
+      .andDo(^(NSInvocation* invocation) {
+        NSString* name;
+        id args;
+        FlutterResult result;
+        [invocation getArgument:&name atIndex:2];
+        [invocation getArgument:&args atIndex:3];
+        [invocation getArgument:&result atIndex:4];
+        FlutterMethodCall* methodCall = [FlutterMethodCall methodCallWithMethodName:name
+                                                                          arguments:args];
+        [weakPlugin handleMethodCall:methodCall result:result];
+      });
+  self.mockTextChecker = [[MockTextChecker alloc] init];
+  id partialMockPlugin = OCMPartialMock(self.plugin);
+  OCMStub([partialMockPlugin textChecker]).andReturn(self.mockTextChecker);
+}
+
+- (void)tearDown {
+  self.plugin = nil;
+  [super tearDown];
+}
+
+#pragma mark - Tests
+
+- (void)testFindAllSpellCheckSuggestionsForText {
+  id textCheckerClassMock = OCMClassMock([UITextChecker class]);
+  [[[textCheckerClassMock stub] andReturn:@[ @"en" ]] availableLanguages];
+  NSArray* suggestions1 = @[ @"suggestion 1", @"suggestion 2" ];
+  NSArray* suggestions2 = @[ @"suggestion 3", @"suggestion 4" ];
+  [self mockUITextCheckerWithExpectedMisspelledWordRange:NSMakeRange(0, 5)
```

Comment:
It seems like you're mostly testing out your mocks here.  Are you sure it's thoroughly testing the corner cases of the range math, like the last part of the string being misspelled, that every letter is included in the ranges, empty strings?

Reply:
> like the last part of the string being misspelled
Example: "This is missspelled" where the last word is being tested.

> that every letter is included in the ranges
Example "missspelled" where the entire string is one misspelled word.